src/build.ts

```diff
@@ -81,14 +82,21 @@ export async function buildModule (opts: BuildModuleOptions) {
 
 async function writeTypes (distDir: string, meta: ModuleMeta) {
   const dtsFile = resolve(distDir, 'types.d.ts')
+  const dtsFileMts = resolve(distDir, 'types.d.mts')
   if (existsSync(dtsFile)) {
     return
   }
 
   // Read generated module types
   const moduleTypesFile = resolve(distDir, 'module.d.ts')
   const moduleTypes = await fsp.readFile(moduleTypesFile, 'utf8').catch(() => '')
-  const typeExports = findExports(moduleTypes)
+  const typeExports = findExports(
+    // Replace `export { type Foo }` with `export { Foo }`
```

Comment:
This is due to https://github.com/Swatinem/rollup-plugin-dts/commit/1b228d052785ea7ad061967430d3ed560582c46c

I am uncertain if such a feature should be included in `mlly` as `findExports` is used to find non-type exports. While `fineTypeExports` finds types exports only. But in `.d.ts` actually, all exports are type exports with or without the `type` indicator.